packages/core/src/messagingProtocols/proteus/ProteusService/ProteusService.ts

```diff
@@ -172,15 +174,15 @@ export class ProteusService {
     }
 
     return this.apiClient.api.conversation.postConversation(payload).catch(async (error: unknown) => {
-      const backendError = error as FederatedBackendsError;
-      switch (backendError.label) {
+      if (!isFederatedBackendsError(error)) {
+        throw error;
+      }
+      switch (error.label) {
         case FederatedBackendsErrorLabel.UNREACHABLE_BACKENDS: {
-          const {backends} = backendError;
-          const users = payload.qualified_users ?? [];
-          const availableUsers: QualifiedId[] = [];
-          const unreachableUsers: QualifiedId[] = [];
-          users.forEach(user =>
-            backends.includes(user.domain) ? unreachableUsers.push(user) : availableUsers.push(user),
+          const {backends} = error;
+          const {excludedUsers: unreachableUsers, includedUsers: availableUsers} = filterUsersFromDomains(
+            payload.qualified_users ?? [],
+            backends,
```

Comment:
I extracted this logic to an external handler and used it in the àddUsersToConversation` method as well :)